meson.build

```diff
@@ -129,170 +129,207 @@ thread_dep = dependency('threads') # pthread for tensorflow-lite
-  torch_dep = dependency('pytorch', required: true)
-
-  if torch_dep.found()
-    add_project_arguments('-DENABLE_PYTORCH=1', language: ['c', 'cpp'])
-  else
-    error('Cannot find pytorch')
+  
+  _variable_postfix = _is_available ? '_when_available' : '_when_not_available'
+  
+  # handle project args
+  project_args += data.get('project_args' + _variable_postfix, {})
+
+  # expose variables for further use
+  _vn = feature_name.underscorify()
+
+  set_variable(_vn + '_deps', _deps)
+  set_variable(_vn + '_is_available', _is_available)
+
+  foreach name, value : data.get('extra_args' + _variable_postfix, {})
+    set_variable(_vn + '_' + name, value)
+  endforeach
+
+  # handle_msg and halt with error when not_available and not_enabled at the sametime by definition
+  _feature_status = _is_available ? 'available' : 'unavailable'
+
+  _msg = data.get('message' + _variable_postfix, '@0@ is @1@.'.format(feature_name, _feature_status))
+  if _is_available
+    message(_msg)
+  else # not available
+    get_option(feature_name).enabled() ? error(_msg) : warning(_msg)
   endif
-endif
+endforeach
 
-# Caffe2
-if get_option('enable-caffe2')
-  caffe2_dep = dependency('caffe2', required: true)
 
-  if caffe2_dep.found() and protobuf_dep.found()
-    add_project_arguments('-DENABLE_CAFFE2=1', language: ['c', 'cpp'])
-  else
-    error('Cannot find caffe2')
-  endif
-endif
+#Definitions enabled by meson_options.txt 
+message('Following project_args are going to be included')
+message(project_args)
+foreach name, value: project_args
+  add_project_arguments('-D@0@=@1@'.format(name, value), language: ['c', 'cpp'])
+endforeach
 
 # Python
 have_python2 = false
 have_python3 = false
 
-if get_option('enable-python')
-  pg_pkgconfig = find_program('pkg-config')
-  # Check python 2.7
-  python2_dep = dependency('python-2.7', required: false)
-  if python2_dep.found()
-    python2_inc_args = []
-    python2_inc_args += run_command(pg_pkgconfig, ['python-2.7', '--cflags']).stdout().strip().split()
-    python2_inc_args += run_command('python2.7', ['-c', 'import site\nfor i in site.getsitepackages(): print("-I" + i + "/numpy/core/include")']).stdout().strip().split()
-    python2_inc_args += '-I' + run_command('python2.7', ['-m', 'site', '--user-site']).stdout().strip() + '/numpy/core/include'
-    python2_inc_valid_args = []
-    foreach python2_inc_arg : python2_inc_args
-      if cxx.has_argument(python2_inc_arg) and \
-          cxx.check_header('numpy/arrayobject.h', args: python2_inc_arg, dependencies : python2_dep)
-        python2_inc_valid_args += python2_inc_arg
-        have_python2 = true
-        break
-      endif
-    endforeach
-    if have_python2
-      python2_deps = declare_dependency(dependencies: python2_dep,
-          compile_args : python2_inc_valid_args)
-    else
-      warning('Found python2.7, but failed to find numpy.')
-      warning('Disable nnstreamer-python2.')
-    endif
-  endif
+pg_pkgconfig = find_program('pkg-config')
```

Comment:
Do you need to put python-dep refactoring in the same PR?

Reply:
python-deps can be put though, I'll save it for another PR (since this pr is too long)